pkg/controller/logstorage/linseed/eskubecontrollers.go

```diff
@@ -29,37 +30,54 @@ import (
 	"github.com/tigera/operator/pkg/render"
 	"github.com/tigera/operator/pkg/render/kubecontrollers"
 	"github.com/tigera/operator/pkg/tls/certificatemanagement"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"k8s.io/apimachinery/pkg/api/errors"
 )
 
-func (r *ReconcileLogStorage) createESKubeControllers(
+// TODO: Move this out of this controller.
+func (r *LinseedSubController) createESKubeControllers(
+	ctx context.Context,
+	request octrl.Request,
 	install *operatorv1.InstallationSpec,
 	hdler utils.ComponentHandler,
 	reqLogger logr.Logger,
 	managementCluster *operatorv1.ManagementCluster,
-	authentication *operatorv1.Authentication,
-	esLicenseType render.ElasticsearchLicenseType,
-	ctx context.Context,
-) (reconcile.Result, bool, error) {
-	kubeControllersUserSecret, err := utils.GetSecret(ctx, r.client, kubecontrollers.ElasticsearchKubeControllersUserSecret, common.OperatorNamespace())
+) error {
+	// Get the Authentication resource.
+	authentication, err := utils.GetAuthentication(ctx, r.client)
+	if err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error while fetching Authentication", err, reqLogger)
+		return err
+	}
+
+	// Get secrets needed for kube-controllers to talk to elastic.
+	kubeControllersUserSecret, err := utils.GetSecret(ctx, r.client, kubecontrollers.ElasticsearchKubeControllersUserSecret, request.TruthNamespace())
 	if err != nil {
 		r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get kube controllers gateway secret", err, reqLogger)
-		return reconcile.Result{}, false, err
+		return err
 	}
 
-	certificateManager, err := certificatemanager.Create(r.client, install, r.clusterDomain, common.OperatorNamespace())
+	certificateManager, err := certificatemanager.Create(r.client, install, r.clusterDomain, request.TruthNamespace())
 	if err != nil {
 		r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the Tigera CA", err, reqLogger)
-		return reconcile.Result{}, false, err
+		return err
 	}
 
 	var managerInternalTLSSecret certificatemanagement.KeyPairInterface
 	if managementCluster != nil {
+		// TODO: This should be moved out of this controller - we shouldn't be creating secrets here, just querying pre-created ones.
 		svcDNSNames := append(dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain), render.ManagerServiceIP)
 		managerInternalTLSSecret, err = certificateManager.GetOrCreateKeyPair(r.client, render.ManagerInternalTLSSecretName, common.CalicoNamespace, svcDNSNames)
 		if err != nil {
 			r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("Error ensuring internal manager TLS certificate %q exists and has valid DNS names", render.ManagerInternalTLSSecretName), err, reqLogger)
-			return reconcile.Result{}, false, err
+			return err
+		}
+	}
+
+	namespaces := []string{request.InstallNamespace()}
```

Comment:
I'm not 100% on this approach yet, but it's a pattern I've been rolling out in this PR.

Some components have ClusterRoles that need to be bound to a namespaced service account. For single-tenant scenarios, this is simply - there's a one-to-one mapping.

For multi-tenant scenarios, we need to bind that ClusterRole to every serviceaccount for the service across all tenants. Essentially, we're sharing the ClusterRole and ClusterRoleBinding for all tenants (since the permissions are identical). 

This code determines which namespaces we need to bind to, and passes them to the render package. By default, we just bind to the namespace we're installing into. But for multi-tenant setups, we query all tenants and discover the full set of namespaces that need to be bound to.